tools/mo/openvino/tools/mo/graph/port.py

```diff
@@ -306,6 +329,31 @@ def get_tensor_debug_info_from_attrs(attrs):
                 fw_debug_info += get_tensor_debug_info_from_attrs(out_node.attrs())
         return fw_debug_info
 
+    def set_tensor_debug_info(self, tensor_info: list, port_renumber: bool = False):
+        """
+        Gets tensor debug info attribute.
+        :param tensor_info: new tensor debug info value.
+        :param port_renumber: defines whether data node index should be calculated considering port renumbering.
+        """
+
+        assert self.type != 'in', "Can't get tensor debug info for input port at {} node".format(self.node.name)
+
+        if self.node.graph.stage == 'front':
+            if self.idx in self.node.out_edges():
+                out_edge = self.node.out_edge(self.idx)
+                out_edge['fw_tensor_debug_info'] = tensor_info
+        else:
+            # before port renumbering we use sequential numbering
+            node_idx = self.idx
+            if port_renumber:
+                if self.node.type != 'Const':
+                    # after port renumbering port indices start from zero,
+                    # but data node indices remain the same
+                    node_idx = self.idx + len(self.node.in_nodes())
+
+            if node_idx in self.node.out_nodes():
+                out_node = self.node.out_node(node_idx)
+                out_node.attrs()['fw_tensor_debug_info'] = tensor_info
```

Comment:
does attr() provide a copy or a reference to the attributes. in other words shouldn't we use 
`out_node['fw_tensor_debug_info']`?

Reply:
attrs() provides reference, both variants are applicable here. 
Changed to out_node['fw_tensor_debug_info'] to align with other usages of fw_tensor_debug_info.